src/full/Agda/TypeChecking/SizedTypes/Syntax.hs

```diff
@@ -170,14 +175,19 @@ instance Plus (SizeExpr' r f) Offset (SizeExpr' r f) where
       Flex x  n -> Flex x  $ n + m
       Infty     -> Infty
 
+-- | Error messages produced by the constraint simplification monad.
+
+type Error = TCM Doc
```

Comment:
Mmh, this does not seem to work in the context of `size-solver`.

I thought about alternatives back and forth, and the best seems to be a structured `Error` type with one constructor for each error.  Then, printing can be handled by the client, either using `PrettyTCM` or something simpler.